atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1058,6 +1059,26 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
         getMeter(AtlasDbMetricNames.SNAPSHOT_TRANSACTION_CELLS_RETURNED, tableRef).mark(resultCount.get());
     }
 
+    /**
+     * A sentinel becomes orphaned if the table has been truncated between the time where the write occurred
+     * and where it was truncated. In this case, there is a chance that we end up with a sentinel with no
+     * valid AtlasDB cell covering it. In this case, we ignore it.
+     */
+    private Set<Cell> findOrphanedSentinels(TableReference table, Map<Cell, Value> rawResults) {
```

Comment:
Are you able to explain why it is okay to ignore the orphaned sentinels? 

What I've pieced together from onboarding and reading code:
* Long running read transaction
* sweep is transactional
* kvs.truncate/drop is not
* if you weren't doing non-transactional things, you wouldn't ever reach this case, even with sweep sweeping

Reply:
if I've reached the sweep sentinel means the world shifted underneath me non-transactionally

if it's not orphaned, I'm reading at too old a transaction
if it is orphaned, this sentinel was written erroneously as part of a truncate/drop, and if you had read it after the truncate you would read nothing => safe to ignore?

Reply:
can we clear up these sweep sentinels? if it was just targeted sweep, we could (not sure if it's a good idea), do a cleanup of those cells by adding on the sweep queue? it would require some special logic I suppose, so not sure if it's worth adding it there or just doing a sweep sentinel deletion here as a best effort

is a sweep sentinel ever deleted? if I've understood correctly, once a cell is swept conservatively it will always have a sweep sentinel until the next non-transactional destructive action?